java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPConnThread.java

```diff
@@ -721,7 +721,7 @@ private synchronized void cacheSearchResult (LDAPSearchListener l, LDAPMessage m
             v = m_messages.get(messageID);
             if (v == null) {
                 m_messages.put(messageID, v = new Vector<Object>());
-                v.addElement(new Long(0));
+                v.addElement(0);
```

Comment:
This needs to be `0L` or we have changed type to `Integer` and will get `ClassCastException` on line 728.